core/src/main/scala/io/qbeast/spark/index/OTreeDataAnalyzer.scala

```diff
@@ -272,13 +213,13 @@ object DoublePassOTreeDataAnalyzer extends OTreeDataAnalyzer with Serializable w
 
   override def analyze(
       dataFrame: DataFrame,
-      indexStatus: IndexStatus): (DataFrame, TableChanges) = {
+      indexStatus: IndexStatus,
+      options: QbeastOptions): (DataFrame, TableChanges) = {
     logTrace(s"Begin: Analyzing the input data with existing revision: ${indexStatus.revision}")
-    // Compute the statistics for the indexedColumns
-    val dataFrameStats = getDataFrameStats(dataFrame, indexStatus.revision.columnTransformers)
-    val numElements = dataFrameStats.getAs[Long]("count")
-    val spaceChanges = calculateRevisionChanges(dataFrameStats, indexStatus.revision)
-    val (isNewRevision, revisionToUse) = spaceChanges match {
+    // Compute the changes in the space: cube size, transformers, and transformations.
+    val (revisionChanges, numElements) =
+      computeRevisionChanges(indexStatus.revision, options, dataFrame)
```

Comment:
I am not sure if it's better to keep `getDataFrameStats` out of the `computeRevisionChanges`, especially if `numElements` is extracted from there.

Reply:
If we take that out, the code gets messy, as it requires the updated Transformers.
We can make updateTransformers public, but it loses encapsulation, and we will repeat some calculations.